streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamTaskTest.java

```diff
@@ -156,23 +166,20 @@ private StreamsConfig createConfig(final boolean enableEoS) throws IOException {
     @Before
     public void setup() throws IOException {
         consumer.assign(Arrays.asList(partition1, partition2));
-        source1.addChild(processorStreamTime);
-        source2.addChild(processorStreamTime);
-        source1.addChild(processorSystemTime);
-        source2.addChild(processorSystemTime);
         config = createConfig(false);
         eosConfig = createConfig(true);
         stateDirectory = new StateDirectory(config, new MockTime());
-        task = new StreamTask(taskId00, partitions, topology, consumer,
-                              changelogReader, config, streamsMetrics, stateDirectory, null, time, producer);
-        task.initialize();
     }
 
     @After
     public void cleanup() throws IOException {
         try {
             if (task != null) {
-                task.close(true, false);
+                try {
+                    task.close(true, false);
+                } catch (Exception e) {
```

Comment:
Why this?

Reply:
For the task that's created to be thrown when close, we can safely sallow it here.

Reply:
But in this test, the task should never throw. Can it? And if yes, would this not be a test failure?